src/canvas/Geometry/Segment.tsx

```diff
@@ -58,4 +66,11 @@ export abstract class Segment {
     abstract isPointNearSegment (point: Point, threshold: number): Point | null;
 
     abstract drawTo(path: Path2D): void;
+
+    abstract getOffsetSegments(distance: number): Segment[];
+
+    abstract translate(displacement: Vector): void;
+
+    abstract clone(): Segment;
+    abstract reversedClone(): Segment;
```

Comment:
It is a clone of the segment, but where the direction is reversed: the start becomes the end, the end becomes the start. This allows us to reverse paths when we create PatternPieces, so that all patternPieces have all edges going around them in the same rotation direction. This is important so that we can trim allowances correctly later.